### PR TITLE
Fix magic-wormhole

### PR DESCRIPTION
**Pull request by jluttine**
###### Motivation for this change

Fixes magic-wormhole by adding some required dependencies and making some necessary upgrades to some packages.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by jluttine** [comment]
Tested the changes in this branch with `nix-shell -p nox --run "nox-review wip --against HEAD~4"`, but it seems there are some build errors not related to this change, those packages were broken anyway. Not sure if there are some build errors actually caused by these changes.

How can there be so many broken packages? Shouldn't nox-review provide quite good protection against it? I guess packages can break if there happens some conflicting changes in separate branches and then they are both merged. But these failures seem like the package definition is broken itself.

@FRidh Btw, you had upgraded magic-wormhole package from 0.9.2 to 0.10.2. Did you actually manage to build it? It broke for me at least, but this should fix it.

File: pkgs/development/python-modules/GeoIP/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{lib, buildPythonPackage, fetchPypi, isPy3k, incremental, ipaddress, twisted
+, automat, zope_interface, idna, pyopenssl, service-identity, pytest, mock, lsof
+, geoip, nose}:
+
+buildPythonPackage rec {
+  name = "${pname}-${version}";
+  pname = "GeoIP";
+  version = "1.3.2";
+
+  buildInputs = [ nose ];
```
Comment (FRidh) [inline pkgs/development/python-modules/GeoIP/default.nix]: checkInputs

File: pkgs/development/python-modules/txtorcon/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{lib, buildPythonPackage, fetchPypi, isPy3k, incremental, ipaddress, twisted
+, automat, zope_interface, idna, pyopenssl, service-identity, pytest, mock, lsof
+, GeoIP}:
+
+buildPythonPackage rec {
+  name = "${pname}-${version}";
+  pname = "txtorcon";
+  version = "0.19.3";
+
+  buildInputs = [ pytest mock lsof GeoIP ];
```
Comment (FRidh) [inline pkgs/development/python-modules/txtorcon/default.nix]: some if not all can go in `checkInputs`

File: pkgs/development/python-modules/txtorcon/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1za4qag4g2lbw695v4ssxqc2aspdyknnbn2diylwg8q9g5k9cczp";
+  };
+
+  # ipaddress isn't required for Python 3 although it's in requirements.txt.
+  # Because ipaddress doesn't install on Python 3, remove the requirement so the
+  # installation of this package doesn't fail on Python 3.
+  patchPhase = "" + lib.optionalString isPy3k ''
```
Comment (FRidh) [inline pkgs/development/python-modules/txtorcon/default.nix]: postPatch

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -680,15 +680,15 @@ in {
 
   attrs = buildPythonPackage (rec {
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: could you move this to a separate expression

**Review by FRidh — CHANGES_REQUESTED** [review]
> How can there be so many broken packages? Shouldn't nox-review provide quite good protection against it? I guess packages can break if there happens some conflicting changes in separate branches and then they are both merged. But these failures seem like the package definition is broken itself.

Occasionally we have mass-rebuilds, and which require fixups by many maintainers.

> Btw, you had upgraded magic-wormhole package from 0.9.2 to 0.10.2. Did you actually manage to build it? It broke for me at least, but this should fix it.

No. I've begun with bumping lots of Python packages, in an effort to get a more up to date package set.
I don't test/fix all of them because I don't have the time for that. This will however distribute the work over other maintainers, as intended ;)

**Comment by jluttine** [comment]
Thanks for the feedback, I'll make those fixes.

About the package upgrades, I understand you, but I don't think that's a good approach. I believe it makes the total workload much greater for everybody. Now I have a problem that I can't test my changes to package X because some package Y is in a broken state. And when I try to fix Y, I can't do it because some other package Z is also broken. I think it would be much less work for everybody if the person writing the package upgrade also made sure it worked (to some extent at least). If someone doesn't have time for that, perhaps he/she can ask others. For instance, the actual package maintainer probably and if none of those, some other group of general package maintainers. Or if there's a lot packages to upgrade, perhaps some call for that: one issue listing everything that should be upgraded and then pinging everyone that's a potential volunteer and then the work can be shared in that issue. Or something better. Anyway, I deeply appreciate your significant efforts for nixpkgs.

**Comment by jluttine** [comment]
I did the fixes you suggested.

**Comment by jluttine** [comment]
Regarding the bunch of other broken Python packages. I can take a look at them at some point. But is there a way to know if someone else working on the same packages? Doesn't make sense for many people to work on fixing the same packages..

**Comment by FRidh** [comment]
Unless people make a mention of it in a issue or PR, no. In any case, it doesn't happen often that people work on fixing the same (Python) packages.

Please, base your work on https://github.com/NixOS/nixpkgs/pull/28884 and open a PR to that PR. I intend to get everything in in one big merge and then merge it into 17.09 as well.

**Comment by jluttine** [comment]
Nice, I'll do that. Thanks.

**Pull request closed without merging**